src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/DeferredBuildTest.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NUnit.Framework;
+using System.IO;
+using Xamarin.ProjectTools;
+
+namespace Xamarin.Android.Build.Tests
+{
+	[Parallelizable (ParallelScope.Children)]
+	public class DeferredBuildTest : BaseTest
+	{
+		[Test]
+		public void SelectivelyRunUpdateAndroidResources ()
+		{
+			var path = Path.Combine ("temp", TestName);
+			var app = new XamarinAndroidApplicationProject {
+				ProjectName = "MyApp",
+			};
+
+			app.SetProperty ("AndroidUseManagedDesignTimeResourceGenerator", "True");
+			app.SetProperty ("AndroidUseIntermediateDesignerFile", "True");
+
+			using (var appBuilder = CreateApkBuilder (Path.Combine (path, app.ProjectName))) {
+				appBuilder.Target = "Compile";
+				Assert.IsTrue (appBuilder.Build (app, parameters: new string[]{
```

Comment:
Same, I think this could use `DesignTimeBuild()`.

Reply:
also, `DesignTimeBuild` doesn't allow me to pass in additional properties either... (which I need to simulate the various combinations of supported/unsupported/deferred/not-deferred)

Reply:
Let's add an overload for that, then. That would be great!

Reply:
Done!